astropy/table/table_helpers.py

```diff
@@ -168,16 +168,16 @@ class ArrayWrapper:
     """
     info = ArrayWrapperInfo()
 
-    def __init__(self, data):
-        self.data = np.array(data)
+    def __init__(self, data, copy=True):
+        self.data = np.array(data, copy=False)
         if 'info' in getattr(data, '__dict__', ()):
             self.info = data.info
 
     def __getitem__(self, item):
         if isinstance(item, (int, np.integer)):
             out = self.data[item]
         else:
-            out = self.__class__(self.data[item])
+            out = self.__class__(self.data[item], copy=False)
```

Comment:
This change is to ensure `ArrayWrapper` also shares memory when being sliced; I think it is OK for this to be backported, but obviously can special-case it if that's felt to be too much of a change.